go/tools/testwrapper/wrap.go

```diff
@@ -0,0 +1,31 @@
+package testwrapper
+
+import (
+	"bytes"
+	"io"
+	"os"
+
+	"os/exec"
+)
+
+// WrapIfNecessary wraps the execution of the test
+func WrapIfNecessary() {
+	if os.Getenv("RAN_FROM_BAZEL") != "1" {
+		return
+	}
```

Comment:
Maybe we should add an override to turn it off, in case the wrapper doesn't work for them for any reason (incompatible  code in `init()`, tests being dependent  on `-test.v`/`t.Verbose`). Maybe a global env variable `RULES_GO_NO_TESTWRAP` settable with `--test_env=`?

Reply:
There should be an environment variable that turns this on or off, so that if someone runs the test binary directly, it doesn't trigger the wrapping. If this should be on default (probably yes?), let's pick something related set by Bazel's test wrapper like `XML_OUTPUT_FILE`.

There should be another environment variable that overrides this. The wrapper can set this in the child process to prevent recursion. Maybe `GO_TEST_WRAP=0`? This could also be used to explicitly turn wrapping on, even when the executable is not run by Bazel.

```
if wrapEnv, ok := os.LookupEnv("GO_TEST_WRAP"); ok {
  wrap, err := strconv.ParseBool(wrapEnv)
  if err != nil {
    log.Fatalf("invalid value for GO_TEST_WRAP: %q", wrapEnv)
  }
  // explicitly wrap or not
}
```